### PR TITLE
Add Resource and Datasource for Netflow Record Policy

### LINKED ISSUE #1210: Add Resource and Datasource for netflowRecordPol.
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

* Missing Resources and Datasources for the netfow record policy (class: netflowRecordPol).

### New or Affected Resource(s) + ACI Class(es):

* aci_netflow_record_policy + netflow:RecordPol

### APIC version and APIC Platform

* V x.x.x and on-prem/cloud-aws/cloud-azure/all.

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by gmicol**
Resolves #1210

### CODE REVIEW

File: docs/data-sources/netflow_record_policy.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  - Default: `uni/infra`
+  
+* `name` (name) - (string) The name of the Netflow Record Policy object.
+
+### Read-Only ###
+
+* `id` - (string) The distinguished name (DN) of the Netflow Record Policy object.
+* `annotation` (annotation) - (string) The annotation of the Netflow Record Policy object.
+* `collect` (collect) - (list) Collect parameters for the flow record.
```
Comment (akinross) [inline docs/data-sources/netflow_record_policy.md:60]: should we change the attribute name to match UI?
```suggestion
* `collect_parameters` (collect) - (list) Collect parameters for the flow record.
```

File: docs/data-sources/netflow_record_policy.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+* `description` (descr) - (string) The description of the Netflow Record Policy object.
+* `match` (match) - (list) Match parameters for the flow record.
```
Comment (akinross) [inline docs/data-sources/netflow_record_policy.md:62]:
```suggestion
* `match_parameters` (match) - (list) Match parameters for the flow record.
```

File: docs/data-sources/netflow_record_policy.md (hunk 1 above)
Comment (akinross) [inline docs/data-sources/netflow_record_policy.md:60]: Should we change the end of the sentence for consistency?
```suggestion
* `collect` (collect) - (list) Collect parameters of the Netflow Record Policy object.
```

File: docs/data-sources/netflow_record_policy.md (hunk 2 above)
Comment (akinross) [inline docs/data-sources/netflow_record_policy.md:62]: Should we change the end of the sentence for consistency?
```suggestion
* `match` (match) - (list) Match parameters of the Netflow Record Policy object.
```

File: docs/resources/netflow_record_policy.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  
+  * `key` (key) - (string) The key used to uniquely identify this configuration object.
+  * `value` (value) - (string) The value of the property.
+
+## Importing
+
+An existing Netflow Record Policy can be [imported](https://www.terraform.io/docs/import/index.html) into this resource with its distinguished name (DN), via the following command:
+
+```
+terraform import aci_netflow_record_policy.example_tenant uni/infra/recordpol-{name}
```
Comment (akinross) [inline docs/resources/netflow_record_policy.md:130]: How does the import work with default? Also should we provide the tenant in the example?

**Review by akinross — CHANGES_REQUESTED** [review]

**Review by akinross — APPROVED** [review]
LGTM

File: gen/templates/resource_test.go.tmpl [hunk 4]
```diff
@@ -33,7 +33,17 @@ func TestAccResource{{$.resourceClassName}}With{{capitalize .class_name}}(t *tes
 			{{- end}}
 		{{- end}}
 		{{- range $key, $value := $.default}}
+			{{- if lt -1 (determineLength $value)}}
```
[the remark this answers is not in the document]
Comment (gmicol) [reply gen/templates/resource_test.go.tmpl:36]: You're definitely right, thanks!

File: gen/templates/testvars.yaml.tmpl [hunk 5]
```diff
@@ -5,9 +5,29 @@
 default:
   {{- range .Properties}}{{$overwritePropertyName := overwriteProperty .PkgName .SnakeCaseName $.Definitions}}
     {{- if .DefaultValue}}
+      {{- if eq .ValueType "bitmask"}}
+        {{- if lt 0 (determineLength (overwritePropertyValue .PkgName $overwritePropertyName .DefaultValue "default" 0 $.Definitions))}}
```
[the remark this answers is not in the document]
Comment (gmicol) [reply gen/templates/testvars.yaml.tmpl:9]: I am not so sure the .ValueType is already a bitmask though. I will check the logic again.

**Review by shrsr — DISMISSED** [review]
LGTM

**Review by samiib — DISMISSED** [review]
LGTM!

**Review by akinross — DISMISSED** [review]
LGTM

**Review by anvitha-jain — APPROVED** [review]

File: docs/resources/netflow_record_policy.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+* `parent_dn` - (string) The distinguished name (DN) of the parent object, possible resources:
+  - [aci_tenant](https://registry.terraform.io/providers/CiscoDevNet/aci/latest/docs/resources/tenant) ([fvTenant](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/fvTenant/overview))
+  - The distinquised name (DN) of classes below can be used but currently there is no available resource for it:
+    - [infraInfra](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/infraInfra/overview)
+
+  - Default: `uni/infra`
+  
+* `annotation` (annotation) - (string) The annotation of the Netflow Record Policy object.
+  - Default: `orchestrator:terraform`
+* `collect_paramaters` (collect) - (list) Collect paramaters of the Netflow Record Policy object.
```
Comment (lhercot) [inline docs/resources/netflow_record_policy.md:101]:
```suggestion
* `collect_parameters` (collect) - (list) Collect parameters of the Netflow Record Policy object.
```

File: gen/definitions/properties.yaml [hunk 7]
```diff
@@ -289,3 +289,41 @@ mplsNodeSidP:
     - class_name: "l3extLoopBackIfP"
       parent_dependency: "l3extRsNodeL3OutAtt"
       parent_dn: "aci_logical_node_to_fabric_node.test.id"
+
+netflowRecordPol:
+  overwrites:
+    match: "match_parameters"
+    collect: "collect_paramaters"
```
Comment (lhercot) [inline gen/definitions/properties.yaml:296]:
```suggestion
    collect: "collect_parameters"
```

File: gen/definitions/properties.yaml [hunk 8]
```diff
@@ -289,3 +289,41 @@ mplsNodeSidP:
...
+  documentation:
+    match: "Match parameters of the %s object."
+    collect: "Collect paramaters of the %s object."
```
Comment (lhercot) [inline gen/definitions/properties.yaml:299]:
```suggestion
    collect: "Collect parameters of the %s object."
```

Comment (lhercot) [reply gen/definitions/properties.yaml:299]: fix same everywhere needed

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by anvitha-jain — APPROVED** [review]

**Review by samiib — APPROVED** [review]
LGTM

**Review by shrsr — APPROVED** [review]
LGTM

**Review by akinross — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**